### ISSUE TITLE
--stats flag ignored as environment variable when --stats-progress is set

### ISSUE THREAD

**Issue by SPDurkee** [rclone/rclone#3098]
#### What is the problem you are having with rclone?
export RCLONE_PROGRESS=true
export RCLONE_STATS=.2s

The stats interval is ignored and defaults to .5s.
If RCLONE_PROGRESS is not set then the stats interval set in the environment variable is utilized correctly

Adding --stats=.2s to the command does work as intended, so the issue is only related to the environment variable and only when the progress environment variable is set.

#### What is your rclone version (output from `rclone version`)
rclone v1.46.0-112-g1c301f9f-beta
- os/arch: linux/amd64
- go version: go1.12.1
2019/04/05 21:03:07 DEBUG : rclone: Version "v1.46.0-112-g1c301f9f-beta" finishing with parameters ["rclone" "version"]

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Ubuntu 16.04.6 LTS 64-bit

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone --transfers=1 copy Temp:folname localfolname
With the following environment variables:
RCLONE_DRIVE_CHUNK_SIZE="256K"
RCLONE_FAST_LIST="true"
RCLONE_LOG_LEVEL="DEBUG"
RCLONE_PROGRESS="true"
RCLONE_STATS=".1s"
RCLONE_STATS_FILE_NAME_LENGTH="0"
RCLONE_STATS_UNIT="bits"
RCLONE_TPSLIMIT="2"

This command does work correctly:
rclone --stats=.1s --transfers=1 copy Temp:folname localfolname

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
N/A as it won't show the update speed of the stats, however I tested this on multiple installations and this is easy to replicate.

**Comment by ncw** [comment]
Ah this bug is here I think

https://github.com/ncw/rclone/blob/0862dc9b2b3ad081ee8fe0152cccef67a2eff907/cmd/cmd.go#L213-L216

This needs to check if the environment variable is set too.

We use this technique in more than one place

```
$ git grep pflag.Lookup
cmd/cmd.go:     statsIntervalFlag := pflag.Lookup("stats")
cmd/rc/rc.go:   if rcFlag := pflag.Lookup(flagName); rcFlag != nil && rcFlag.Changed {
fs/config/configflags/configflags.go:   logLevelFlag := pflag.Lookup("log-level")
fs/config/flags/flags.go:               flag := pflag.Lookup(name)
```

So I think what is needed is a a routine in `fs/config/flags/flags.go` say `IsPresent` which does essentially what `ShowStats` does but for an arbitrary variable, **and** checks the environment to see if it is present.

Fancy having a go at this?

**Comment by SPDurkee** [comment]
After moving close to 1PB of data, I certainly have used rclone enough to warrant spending some time fixing bugs. I have 30 years of coding experience, but haven't done anything in Go yet. I don't mind giving it a shot. Worst that can happen is my pull request isn't accepted :)

I'll get a development environment setup and start working on it. Will let you know if I run into any roadblocks or have any questions.

**Comment by ncw** [comment]
> After moving close to 1PB of data, I certainly have used rclone enough to warrant spending some time fixing bugs. I have 30 years of coding experience, but haven't done anything in Go yet. I don't mind giving it a shot. Worst that can happen is my pull request isn't accepted :)
> 
> I'll get a development environment setup and start working on it. Will let you know if I run into any roadblocks or have any questions.

That would be super! Drop me a note if you get stuck or need help.

**Comment by ncw** [comment]
This is now fixed in v1.50
